docs/how-to/using-omnitrace-api.rst

```diff
@@ -0,0 +1,268 @@
+
+      omnitrace_user_push_region("thread_wait");
+      for(auto& itr : threads)
+         itr.join();
+      omnitrace_user_pop_region("thread_wait");
+
+      run(nitr, nfib);
+
+      printf("[%s] fibonacci(%li) x %lu = %li\n", argv[0], nfib, nthread, total.load());
+      omnitrace_user_pop_region(argv[0]);
+
+      return 0;
+   }
+
+   long
+   fib(long n)
+   {
+      return (n < 2) ? n : fib(n - 1) + fib(n - 2);
+   }
+
+   #define RUN_LABEL                                                                        \
+      std::string{ std::string{ __FUNCTION__ } + "(" + std::to_string(n) + ") x " +        \
+                  std::to_string(nitr) }                                                  \
+         .c_str()
+
+   void
+   run(size_t nitr, long n)
+   {
+      omnitrace_user_push_region(RUN_LABEL);
+      long local = 0;
+      for(size_t i = 0; i < nitr; ++i)
+         local += fib(n);
+      total += local;
+      omnitrace_user_pop_region(RUN_LABEL);
+   }
+
+   int
+   custom_push_region(const char* name)
+   {
+      if(!original_callbacks.push_region || !original_callbacks.push_annotated_region)
+         return OMNITRACE_USER_ERROR_NO_BINDING;
+
+      printf("Pushing custom region :: %s\n", name);
+
+      if(original_callbacks.push_annotated_region)
+      {
+         int32_t _err = errno;
+         char*   _msg = nullptr;
+         char    _buff[1024];
+         if(_err != 0) _msg = strerror_r(_err, _buff, sizeof(_buff));
+
+         omnitrace_annotation_t _annotations[] = {
+               { "errno", OMNITRACE_INT32, &_err }, { "strerror", OMNITRACE_STRING, _msg }
+         };
+
+         errno = 0;  // reset errno
+         return (*original_callbacks.push_annotated_region)(
+               name, _annotations, sizeof(_annotations) / sizeof(omnitrace_annotation_t));
+      }
+
+      return (*original_callbacks.push_region)(name);
+   }
+
+Output from the API example program
+========================================
+
+First, instrument and run the program.
+
+.. code-block:: shell
+
+   $ omnitrace-instrument -l --min-instructions=8 -E custom_push_region -o -- ./user-api
+   ...
+   $ omnitrace-run --profile --use-pid off --time-output off -- ./user-api.inst 20 4 100
+   Pushing custom region :: ./user-api.inst
+   [omnitrace][omnitrace_init_tooling] Instrumentation mode: Trace
+
+
+         ______   .___  ___. .__   __.  __  .___________..______          ___       ______  _______
+      /  __  \  |   \/   | |  \ |  | |  | |           ||   _  \        /   \     /      ||   ____|
+      |  |  |  | |  \  /  | |   \|  | |  | `---|  |----`|  |_)  |      /  ^  \   |  ,----'|  |__
+      |  |  |  | |  |\/|  | |  . `  | |  |     |  |     |      /      /  /_\  \  |  |     |   __|
+      |  `--'  | |  |  |  | |  |\   | |  |     |  |     |  |\  \----./  _____  \ |  `----.|  |____
+      \______/  |__|  |__| |__| \__| |__|     |__|     | _| `._____/__/     \__\ \______||_______|
```

Comment:
The ASCII art looks off. Not just here, but in the ReadTheDocs preview build. I'm not sure if that's an easy fix with this format, though.

Reply:
Good catch! I think this happened because I indented the entire shell output. I think can fix it so it looks a bit better. I'll work on some of these things tomorrow morning.